### DISCUSSION TITLE
设置上游代理后无法播放音乐，取消正常

### DISCUSSION THREAD

**Discussion by sengoku-f** [1715173329/UnblockNeteaseMusic#768] [Q&A]
如题，使用 youtube 音源要设置代理，但是设置了-u http://127.0.0.1:1080 后无法播放音乐

**Comment by pan93412** [comment]
YouTube 的代理不是在那邊設定的，需要傳 `HTTP_PROXY` 或 `HTTPS_PROXY`。

**Reply by sengoku-f** [reply]
如何设置，能详情说一下吗？ 
`HTTP_PROXY="http://127.0.0.1:1080" node app.js -p 8080:8081 -e https://music.163.com -o ytdlp -s` 这样？

**Reply by 1715173329** [reply]
yup，你可能还要指定 HTTPS_PROXY，以及再用小写传一遍。

**Reply by sengoku-f** [reply]
> yup，你可能还要指定 HTTPS_PROXY，以及再用小写传一遍。

试了还是不能使用，命令`LOG_LEVEL=debug HTTP_PROXY="http://127.0.0.1:1080" HTTPS_PROXY="http://127.0.0.1:1080" http_proxy="http://127.0.0.1:1080" https_proxy="http://127.0.0.1:1080" node app.js -p 8080:8081 -e https://music.163.com -o ytdlp -s`

```
ERROR: (provider/match) connect ETIMEDOUT 199.96.62.75:443
    Error: connect ETIMEDOUT 199.96.62.75:443
        at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1159:16)
    errno: -60
    code: ETIMEDOUT
    syscall: connect
    address: 199.96.62.75
    port: 443
(node:43841) UnhandledPromiseRejectionWarning: TypeError: AggregateError is not a constructor
```

**Reply by LastStranger** [reply]
有解决吗?我也想解决这个问题
